### PR TITLE
Surveys: properly show sections in submission view

### PR DESCRIPTION
**Pull request by pferreir**
We're currently just listing questions/results with no ordering or structure.

### CODE REVIEW

File: indico/MaKaC/webinterface/tpls/events/include/Common.tpl [hunk 1]
```diff
@@ -1,9 +1,5 @@
 <%def name="renderDescription(text)">
-    % if isStringHTML(text):
-        <span itemprop="description">${text}</span>
-    % else:
-        <pre itemprop="description">${text}</pre>
-    % endif
+    <span itemprop="description">${text}</span>
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/tpls/events/include/Common.tpl]: Are you sure you want to include this? It's the commit that broke linebreaks in descriptions.

File: indico/modules/events/surveys/templates/management/survey_submission.html [hunk 2]
```diff
@@ -21,13 +21,26 @@
...
 {% block content %}
-    <div class="i-box-content">
-        <dl class="i-data-list">
-            {% for answer in submission.answers -%}
-                <dt><strong>{{ answer.question.title }}</strong></dt>
-                <dd>{{ _render_answer(answer) }}</dd>
-            {%- endfor %}
-        </dl>
+    <div id="survey-submission" class="i-box-content">
+        {% for section in survey.sections if section.children -%}
```
Comment (ThiefMaster) [inline indico/modules/events/surveys/templates/management/survey_submission.html]: I think the nested loop here will send multiple SQL queries (one for each section to get its children).

Adding this in `RHSurveySubmissionBase` would probably avoid this:

    sections_strategy = defaultload('sections').joinedload('children')

And in the query options:

    .options(answers_strategy, survey_strategy, sections_strategy)

Comment (pferreir) [inline indico/modules/events/surveys/templates/management/survey_submission.html]: This is a `SurveySubmission` object, there are no `sections` in it. I tried `joinedload('survey').defaultload('sections').joinedload('children')`, but it doesn't seem to work either :disappointed:.

Comment (ThiefMaster) [inline indico/modules/events/surveys/templates/management/survey_submission.html]: Hm, that should work - one extra query to get the sections but no looped queries...

Comment (pferreir) [inline indico/modules/events/surveys/templates/management/survey_submission.html]: Yeah, maybe it does, have to have a closer look at the logs. :hamburger:

File: indico/MaKaC/webinterface/tpls/events/include/Common.tpl (hunk 1 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/tpls/events/include/Common.tpl]: Ah yeah indeed. It's just that I forgot to remove it locally.

**Comment by pferreir** [comment]
Merged in c5c799d9c1cc4fd1de5a2553dd38e1bf49bb63d7

**Pull request closed without merging**